src/util.js

```diff
@@ -268,6 +272,33 @@ define([
         pos = pos + text.length;
         that.setCaretPosition(ctrl, start, pos);
     };
+
+
+    that.xmlDiff = function (localForm, serverForm, opts) {
+        function cleanForDiff (value) {
+            // convert leading tabs to spaces
+            value = value.replace(/^\t+/mg, function (match) {
+                return match.replace(/\t/g, "  ");
+            });
+            // add newline at end of file if missing
+            if (!value.match(/\n$/)) {
+                value = value + "\n";
+            }
+            return value;
+        }
+        opts = opts || {};
+        if (opts.normalize_xmlns) {
+            var xmlns = $($.parseXML(serverForm)).find('data').attr('xmlns');
+            localForm = localForm.replace(/(data[^>]+xmlns=")(.+?)"/,
+                                    '$1' + xmlns + '"');
+        }
+        localForm = cleanForDiff(localForm);
+        serverForm = cleanForDiff(serverForm);
+        var patch = jsdiff.createPatch("", localForm, serverForm, "Local Form", "Server Form");
```

Comment:
What do you think of swapping the order of these? Right now it renders as
```
--- Local Form
+++ Server Form
```
but *Local Form* will overwrite *Server Form*. Seems more intuitive to me to show `+` beside lines that will be kept and `-` beside lines that will be removed.